controllers/operatorconfigmap_controller.go

```diff
@@ -706,32 +692,47 @@ func (c *OperatorConfigMapReconciler) reconcileSubscription() error {
 		}
 	}
 
-	if desiredChannel != "" {
-		subscriptions := &opv1a1.SubscriptionList{}
-		err := c.list(
-			subscriptions,
-			client.InNamespace(c.OperatorNamespace),
-			client.MatchingLabels{subscriptionLabelKey: subscriptionLabelValue},
-			client.Limit(1),
-		)
-		if err != nil {
-			return fmt.Errorf("failed to list subscription for ocs-client-operator using labels: %v", err)
-		}
+	clientSubscription, err := c.getSubscriptionByPackageName("ocs-client-operator")
+	if err != nil {
+		return err
+	}
 
-		if len(subscriptions.Items) == 1 {
-			clientSubscription := &subscriptions.Items[0]
-			if desiredChannel != clientSubscription.Spec.Channel {
-				clientSubscription.Spec.Channel = desiredChannel
-				// TODO: https://github.com/red-hat-storage/ocs-client-operator/issues/130
-				// there can be a possibility that platform is behind, even then updating the channel will only make subscription to be in upgrading state
-				// without any side effects for already running workloads. However, this will be a silent failure and need to be fixed via above TODO issue.
-				if err := c.update(clientSubscription); err != nil {
-					return fmt.Errorf("failed to update subscription channel to %v: %v", desiredChannel, err)
-				}
-			}
+	updateRequired := utils.AddLabel(clientSubscription, subscriptionLabelKey, subscriptionLabelValue)
+	if desiredChannel != "" && desiredChannel != clientSubscription.Spec.Channel {
+		clientSubscription.Spec.Channel = desiredChannel
+		// TODO: https://github.com/red-hat-storage/ocs-client-operator/issues/130
+		// there can be a possibility that platform is behind, even then updating the channel will only make subscription to be in upgrading state
+		// without any side effects for already running workloads. However, this will be a silent failure and need to be fixed via above TODO issue.
+		updateRequired = true
+	}
+
+	if updateRequired {
+		if err := c.update(clientSubscription); err != nil {
+			return fmt.Errorf("failed to update subscription channel to %v: %v", desiredChannel, err)
 		}
 	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) reconcileCSIAddonsSubscription() error {
+	clientSubscription, err := c.getSubscriptionByPackageName("ocs-client-operator")
+	if err != nil {
+		return err
+	}
 
+	addonsSubscription, err := c.getSubscriptionByPackageName("odf-csi-addons-operator")
+	if kerrors.IsNotFound(err) {
+		addonsSubscription, err = c.getSubscriptionByPackageName("csi-addons")
+	}
+	if err != nil {
+		return err
+	}
+	if addonsSubscription.Spec.Channel != clientSubscription.Spec.Channel {
+		addonsSubscription.Spec.Channel = clientSubscription.Spec.Channel
+		if err := c.update(addonsSubscription); err != nil {
+			return fmt.Errorf("failed to update subscription channel of 'csi-addons' to %v: %v", addonsSubscription.Spec.Channel, err)
+		}
+	}
```

Comment:
Why use the version on the client subscription and not the desired subscription?

Reply:
done.